### ISSUE TITLE
systemd-nspawn with transparent ID mapping fails with OverlayFS

### ISSUE THREAD

**Issue by mpfau** [systemd/systemd#24542]
### systemd version the issue has been seen with

systemd-251.3-1~bpo11+1

### Used distribution

Debian 11

### Linux kernel version used

5.18.0-0.bpo.1-amd64

### CPU architectures issue was seen on

x86_64

### Component

systemd-nspawn

### Expected behaviour you didn't see

When starting a container with `--private-users-ownership=map` and an overlayfs mount, I would expect it to startup.

### Unexpected behaviour you saw

Startup failed with the following error message:
```
Failed to mount overlay (type overlay) on /mnt/data/machines/test/etc (MS_RDONLY "lowerdir=/mnt/data/machines/test/etc:/etc"): Invalid argument
```

### Steps to reproduce the problem

1. create a test container in `/var/lib/machines`
2. run the following commands

```
cd /var/lib/machines/test
systemd-nspawn --private-users=pick --private-users-ownership=map --boot --overlay-ro=/etc:+/etc
```

Boot should fail with error msg mentioned above.

### Additional program output to the terminal or log subsystem illustrating the issue
Booting and mounting the OverlayFS works fine when using `--private-users=yes`.

```sh
systemd-nspawn --private-users=yes --boot --overlay-ro=/etc:+/etc
```

**Comment by poettering** [comment]
idmap support for overlayfs was only implemented in kernel 5.19

**Comment by mpfau** [comment]
Sorry! I thought it came with 5.12 but that were only bind mounts...
